mono/metadata/loader-internals.h

```diff
@@ -30,9 +39,24 @@ struct _MonoAssemblyLoadContext {
 };
 #endif /* ENABLE_NETCORE */
 
+void
+mono_global_loader_data_lock (void);
+
```

Comment:
Does this need to be exported ? It looks like a lock internal to native-library.c.

Reply:
The lock is in loader.c, where I've tried to keep the more generic loader stuff, but it's called from native-library. 

I can potentially move more stuff over to native-library, but if that function isn't exported I'll have to move `mono_loader_init`, which I was trying to avoid since I don't think it belongs there.